### PR TITLE
Mqtt frames v3

### PR DESCRIPTION
**Pull request by hsadia538**
Ticket: https://redmine.openinfosecfoundation.org/issues/5731

Previous PR: #1065 

Describe changes:
- Add tests for mqtt frames for truncated messages

Requires: https://github.com/OISF/suricata/pull/8513

### CODE REVIEW

**Review by jufajardini — CHANGES_REQUESTED** [review]
I have added a few comments.  :)
Especially for the truncated test checks, I would like us to check on the frame length, too, so we can ensure things are being calculated as expected.
I am again wondering if we should have a special type for truncated frames, as, for instance, the length for the pdu frame in such case isn't the actual pdu length, just the max that suricata will process, as per MQTT standards.
So, having a truncated frame might let people know what to expect in such cases. Maybe in such circumstances, we could have a `trunc_data` frame instead of a `data` frame?

File: tests/mqtt-frames-truncated/test.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+pcap: ../mqtt-limit-1/input.pcap
+
+requires:
```
[the remark this answers is not in the document]
Comment (hsadia538) [reply tests/mqtt-frames-truncated/test.yaml:3]: alright

**Comment by hsadia538** [comment]
> I have added a few comments. :) Especially for the truncated test checks, I would like us to check on the frame length, too, so we can ensure things are being calculated as expected. I am again wondering if we should have a special type for truncated frames, as, for instance, the length for the pdu frame in such case isn't the actual pdu length, just the max that suricata will process, as per MQTT standards. So, having a truncated frame might let people know what to expect in such cases. Maybe in such circumstances, we could have a `trunc_data` frame instead of a `data` frame?

Sure I can do that . Will update this PR after the new frames PR

**Comment by hsadia538** [comment]
> I have added a few comments. :) Especially for the truncated test checks, I would like us to check on the frame length, too, so we can ensure things are being calculated as expected. I am again wondering if we should have a special type for truncated frames, as, for instance, the length for the pdu frame in such case isn't the actual pdu length, just the max that suricata will process, as per MQTT standards. So, having a truncated frame might let people know what to expect in such cases. Maybe in such circumstances, we could have a `trunc_data` frame instead of a `data` frame?

Do you also think that we should Have `trunc_pdu` as well along with `trunc_data` or just `trunc_data`

**Comment by jufajardini** [comment]
> > I have added a few comments. :) Especially for the truncated test checks, I would like us to check on the frame length, too, so we can ensure things are being calculated as expected. I am again wondering if we should have a special type for truncated frames, as, for instance, the length for the pdu frame in such case isn't the actual pdu length, just the max that suricata will process, as per MQTT standards. So, having a truncated frame might let people know what to expect in such cases. Maybe in such circumstances, we could have a `trunc_data` frame instead of a `data` frame?
> 
> Do you also think that we should Have `trunc_pdu` as well along with `trunc_data` or just `trunc_data`

🤔  What about asking the others about this?

**Comment by hsadia538** [comment]
followed by #1123

**Pull request closed without merging**